tchannel/schemes/thrift.py

```diff
@@ -18,9 +18,11 @@ def __init__(self, tchannel):
         self._tchannel = tchannel
 
     @gen.coroutine
-    def __call__(self, request=None, headers=None, timeout=None,
+    def __call__(self, request, headers=None, timeout=None,
                  retry_on=None, retry_limit=None):
 
+        assert request, "a ThriftRequest is required"
```

Comment:
If you want to guard here this should be an isinstance check (I can pass `True`). I'd also recommend a different exception type.